Don't search for MPI in non-MPI builds

**Describe the bug**
<!-- A clear and concise description of what the bug is.-->
The CMake configuration searches always for MPI, which currently leads to failing local builds with Intel oneAPI classic compilers for me.

**To Reproduce**
<!-- Steps to reproduce the behaviour:

1. minimal example of how to trigger the bug (include self-contained input file)
2. which version of DFTB+ is used (downloaded, conda-forge or build from source),
   if built from source include the git commit, the compiler and used libraries
3. output showing the error

All input and output files required to confirm your report. -->

Trying CMake 3.19, Ninja 1.10, Intel oneAPI classic compiler 2021.1 with

```
source /opt/intel/oneapi/setvars.sh
FC=ifort CC=icc cmake -B _build -G Ninja
```

Failes with:

```
-- Reading global build config file: /home/awvwgk/projects/src/git/dftbplus/config.cmake
-- The Fortran compiler identification is Intel 20.2.1.20201112
-- The C compiler identification is Intel 20.2.1.20201112
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /opt/intel/oneapi/compiler/2021.1.1/linux/bin/intel64/ifort - skipped
-- Checking whether /opt/intel/oneapi/compiler/2021.1.1/linux/bin/intel64/ifort supports Fortran 90
-- Checking whether /opt/intel/oneapi/compiler/2021.1.1/linux/bin/intel64/ifort supports Fortran 90 - yes
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/intel/oneapi/compiler/2021.1.1/linux/bin/intel64/icc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Build type: RelWithDebInfo (default single-config)
-- Reading build environment specific toolchain file: /home/awvwgk/projects/src/git/dftbplus/sys/intel.cmake
-- Flags for Fortran-compiler (build type: RelWithDebInfo):  -standard-semantics   -g -O2 -ip
-- Flags for C-compiler (build type: RelWithDebInfo):    
-- FLAGS:  -standard-semantics   -g -O2 -ip
CMake Error in /home/awvwgk/projects/src/git/dftbplus/_build/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Imported target "MPI::MPI_C" includes non-existent path

    "/home/awvwgk/projects/src/git/dftbplus/'/opt/intel/oneapi/mpi/2021.1.1/include'"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error at /usr/share/cmake-3.19/Modules/FindMPI.cmake:1213 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  /usr/share/cmake-3.19/Modules/FindMPI.cmake:1264 (_MPI_try_staged_settings)
  /usr/share/cmake-3.19/Modules/FindMPI.cmake:1529 (_MPI_check_lang_works)
  CMakeLists.txt:33 (find_package)


-- Configuring incomplete, errors occurred!
```

**Expected behaviour**
<!-- A clear and concise description of what you expected to happen, instead of what occurred -->

Don't search for MPI if it is not required for the build.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,14 +30,12 @@ if(WITH_API)
   dftbp_get_api_version(API_VERSION API_VERSION_MAJOR API_VERSION_MINOR API_VERSION_PATCH)
 endif()
 
-find_package(MPI QUIET)
-if(WITH_MPI AND NOT MPI_FORTRAN_FOUND)
-  message(FATAL_ERROR "Compiler ${CMAKE_Fortran_COMPILER} is not MPI capable but is specified for "
-    "a WITH_MPI=TRUE build")
-elseif(NOT WITH_MPI AND MPI_FORTRAN_FOUND)
-  message(WARNING "MPI enabled compiler ${CMAKE_Fortran_COMPILER} found for a non-MPI build. Your "
-    "build will NOT be MPI-parallelised. Set WITH_MPI=TRUE in order to obtain an MPI-parallelised "
-    "build.")
+if(WITH_MPI)
+  find_package(MPI QUIET)
+  if(NOT MPI_FORTRAN_FOUND)
+    message(FATAL_ERROR "Compiler ${CMAKE_Fortran_COMPILER} is not MPI capable but is specified for "
+      "a WITH_MPI=TRUE build")
+  endif()
 endif()
 
 if(WITH_OMP AND NOT TARGET OpenMP::OpenMP_Fortran)

